internal/pkg/aws/ecs/ecs.go

```diff
@@ -99,6 +111,58 @@ func (e *ECS) Service(clusterName, serviceName string) (*Service, error) {
 	return nil, fmt.Errorf("cannot find service %s", serviceName)
 }
 
+// UpdateServiceOpts sets the optional parameter for UpdateService.
+type UpdateServiceOpts func(*ecs.UpdateServiceInput)
+
+// WithForceUpdate sets ForceNewDeployment to force an update.
+func WithForceUpdate() UpdateServiceOpts {
+	return func(in *ecs.UpdateServiceInput) {
+		in.ForceNewDeployment = aws.Bool(true)
+	}
+}
+
+// UpdateService calls ECS API and updates the specific service running in the cluster.
+func (e *ECS) UpdateService(clusterName, serviceName string, opts ...UpdateServiceOpts) error {
+	in := &ecs.UpdateServiceInput{
+		Cluster: aws.String(clusterName),
+		Service: aws.String(serviceName),
+	}
+	for _, opt := range opts {
+		opt(in)
+	}
+	svc, err := e.client.UpdateService(in)
+	if err != nil {
+		return fmt.Errorf("update service %s from cluster %s: %w", serviceName, clusterName, err)
+	}
+	s := Service(*svc.Service)
+	if err := e.waitUntilServiceStable(&s); err != nil {
+		return fmt.Errorf("wait until service %s becomes stable: %w", serviceName, err)
+	}
+	return nil
+}
+
+// waitUntilServiceStable waits until the service is stable.
+// See https://docs.aws.amazon.com/cli/latest/reference/ecs/wait/services-stable.html
+func (e *ECS) waitUntilServiceStable(svc *Service) error {
+	var err error
+	var tryNum int
+	for {
+		if len(svc.Deployments) == stableServiceDeploymentNum &&
+			aws.Int64Value(svc.DesiredCount) == aws.Int64Value(svc.RunningCount) {
```

Comment:
i asked this offline but i'm also putting it here just for the records: do we want to differentiate a stable rolled-back service from a stable updated service?

Reply:
Discussed offline before the rollback it should have already triggered the timeout error. Edited the error log to prompt users to run `svc status` to check the status of their deployment.